src/arrayMethodJoin.js

```diff
@@ -4,8 +4,21 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const splitter = typeof separator === 'string' ? separator : separator + '';
```

Comment:
You can use just String object constructor to avoid additional check, like String()

Reply:
I just wanted to be precise like they all do in algorithms) But okay, I'll change that